PoSSapi/src/Application/Reports/Commands/GenerateReportCommand.cs

```diff
@@ -29,20 +29,26 @@ public GenerateReportCommandHandler(IApplicationDbContext context, IMapper mappe
 
     public async Task<ReportDto> Handle(GenerateReportCommand request, CancellationToken cancellationToken)
     {
-        var entity = new Report();
-        
-        _mapper.Map(request, entity);
+        var entity = new Report {
+            Id = request.Id,
+            StartTime = request.StartTime,
+            EndTime = request.EndTime
+        };
 
-        entity.Revenue = _context.Payments
+        var payments = _context.Payments
             .Where(x => x.TimeWhenCompleted >= request.StartTime)
             .Where(x => x.TimeWhenCompleted <= request.EndTime)
-            .Sum(x => x.PriceOfOrder);
-        
+            .ToList<Payment>();
+
+        entity.Revenue = payments.Sum(x => x.PriceOfOrder)
+            + payments.Sum(x => x.Tip)
+            - payments.Sum(x => x.Discount);
```

Comment:
The `Discount` is probably a decimal or percent value, so it shouldn't be just subtracted. Also the `Tip` belongs to the waiter, so I don't know if it should be added to the revenue

Reply:
You're right, discount is a percentage according to the design document. I will fix that. Also, if tips are not counted as revenue, why are they even stored in the database?

Reply:
I guess a `Tip` is not revenue for the business owner, only for the waiter. I don't have the best answer here, you may leave it as it is probably, because it's a business rule we don't know.